src/lists/list.jsx

```diff
@@ -92,35 +95,32 @@ const List = React.createClass({
       ...other,
     } = this.props;
 
+    let hasSubheader = false;
+
+    if (subheader) {
+      hasSubheader = true;
+    } else {
+      const firstChild = React.Children.toArray(children)[0];
+      if (React.isValidElement(firstChild) && firstChild.type === Subheader) {
```

Comment:
Oh yeah, I'm really interested in removing the use of `displayName` for the check. Then, I would be able to write a plugin that remove the displayName from the production build :grin: